src/modules/Thread/context/ThreadProvider.tsx

```diff
@@ -160,22 +171,41 @@ export const ThreadProvider: React.FC<ThreadProviderProps> = (props: ThreadProvi
     latestMessageTimeStamp: allThreadMessages[allThreadMessages.length - 1]?.createdAt || 0,
   }, { logger, threadDispatcher });
   const toggleReaction = useToggleReactionCallback({ currentChannel }, { logger });
+
+  // Send Message Hooks
   const sendMessage = useSendUserMessageCallback({
     isMentionEnabled,
     currentChannel,
-  }, { logger, pubSub, threadDispatcher });
+    onBeforeSendUserMessage,
+  }, {
+    logger,
+    pubSub,
+    threadDispatcher,
+  });
   const sendFileMessage = useSendFileMessageCallback({
     currentChannel,
-  }, { logger, pubSub, threadDispatcher });
+    onBeforeSendFileMessage,
+  }, {
+    logger,
+    pubSub,
+    threadDispatcher,
+  });
   const sendVoiceMessage = useSendVoiceMessageCallback({
-    currentChannel, onBeforeSendVoiceMessage,
-  }, { logger, pubSub, threadDispatcher });
+    currentChannel,
+    onBeforeSendVoiceMessage,
+  }, {
+    logger,
+    pubSub,
+    threadDispatcher,
+  });
   const [sendMultipleFilesMessage] = useSendMultipleFilesMessage({
     currentChannel,
+    onBeforeSendMultipleFilesMessage,
   }, {
     logger,
     pubSub,
```

Comment:
no `threadDispatcher`?

Reply:
Yes, we only use the `pubSub` instead, inside of the useSendMultipleFilesMessage hook!

Reply:
ok